crypto/evp/p_lib.c

```diff
@@ -1459,3 +1459,142 @@ int evp_pkey_downgrade(EVP_PKEY *pk)
     return 0;     /* No downgrade, but at least the key is restored */
 }
 #endif  /* FIPS_MODE */
+
+const OSSL_PARAM *EVP_PKEY_gettable_params(EVP_PKEY *pkey)
+{
+    if (pkey == NULL
+        || pkey->keymgmt == NULL
+        || pkey->keydata == NULL)
+        return 0;
+    return evp_keymgmt_gettable_params(pkey->keymgmt);
+}
+
+/*
+ * For the following methods param->return_size is set to a value
+ * larger than can be returned by the call to evp_keymgmt_get_params().
+ * If it is still this value then the parameter was ignored - and in this
+ * case it returns an error..
+ */
+
+int EVP_PKEY_get_bn_param(EVP_PKEY *pkey, const char *key_name, BIGNUM **bn)
+{
+    /*
+     * TODO(3.0) make this buffer dynamic.
+     * The evp_keymgmt_get_params() call may fail if the buffer is too small.
+     * We could possibly use the return_size, alloc and call again if this
+     * happens.
+     */
+    unsigned char buffer[2048];
+    OSSL_PARAM params[2];
+    size_t not_set = sizeof(buffer) + 1;
```

Comment:
Use const ? Also would it make sense to use rather (size_t) -1 so it is invariant even if the buffer is eventually made a dynamically allocated one?

Reply:
I will just use const - if we need to handle the dynamic case then it will be probably not be a const. This should do for now.